person: Add default daily rate to person overview table

FrederikSchlemmer opened the pull request:
This pull request extends the person overview table with the field of the defaultDailyRate, which was implemented in PR #413.
For this it was necessary to extend the PersonBaseData and PersonBaseDataBean with this field.

**Note: This Pull Request is based on #413, so first the PR #413 must be merged and then a rebase is necessary.**

Closes #429

czarnecki reviewed (approved):
Looks fine code wise. Didn't test it out, because I don't have the environment set up right now, so I can't comment on that.

tinne commented on budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/persistence/person/PersonRepository.java:
```diff
@@ -11,10 +11,10 @@
     List<PersonEntity> findByProjectIdOrderByNameAsc(long projectId);
 
-    @Query("select new org.wickedsource.budgeteer.persistence.person.PersonBaseDataBean(p.id, p.name, sum(r.minutes * r.dailyRate), sum(r.minutes), max(r.date)) from PersonEntity p left join p.workRecords r where p.project.id=:projectId group by p.id, p.name order by p.name")
+    @Query("select new org.wickedsource.budgeteer.persistence.person.PersonBaseDataBean(p.id, p.name, sum(r.minutes * r.dailyRate), sum(r.minutes), max(r.date), p.defaultDailyRate) from PersonEntity p left join p.workRecords r where p.project.id=:projectId group by p.id, p.name order by p.name")
```
p.defaultDailyRate must be included into the group by clause as well to turn this into a valid query. See patch attached: 
[fix-missing-group-by.patch.txt](https://github.com/adessoAG/budgeteer/files/5486771/fix-missing-group-by.patch.txt)

tinne commented on budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/persistence/person/PersonRepository.java:
```diff
@@ -11,10 +11,10 @@
     List<PersonBaseDataBean> findBaseDataByProjectId(@Param("projectId") long projectId);
 
-    @Query("select new org.wickedsource.budgeteer.persistence.person.PersonBaseDataBean(p.id, p.name, sum(r.minutes * r.dailyRate), sum(r.minutes), max(r.date)) from PersonEntity p left join p.workRecords r where p.id=:personId group by p.id, p.name order by p.name")
+    @Query("select new org.wickedsource.budgeteer.persistence.person.PersonBaseDataBean(p.id, p.name, sum(r.minutes * r.dailyRate), sum(r.minutes), max(r.date), p.defaultDailyRate) from PersonEntity p left join p.workRecords r where p.id=:personId group by p.id, p.name order by p.name")
```
see line 14

tinne commented on budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/persistence/person/PersonRepository.java:
```diff
@@ -27,6 +27,6 @@
     void deleteByProjectId(@Param("projectId") long projectId);
 
-    @Query("select new org.wickedsource.budgeteer.persistence.person.PersonBaseDataBean(p.id, p.name, sum(r.minutes * r.dailyRate), sum(r.minutes), max(r.date)) from PersonEntity p left join p.workRecords r where r.budget.id = :budgetId group by p.id, p.name order by p.name")
+    @Query("select new org.wickedsource.budgeteer.persistence.person.PersonBaseDataBean(p.id, p.name, sum(r.minutes * r.dailyRate), sum(r.minutes), max(r.date), p.defaultDailyRate) from PersonEntity p left join p.workRecords r where r.budget.id = :budgetId group by p.id, p.name order by p.name")
```
see line 14

tinne reviewed (changes requested):
There is a SQL error (thrice) in PersonRepository. See 
[budgeteer - missing GROUP BY.log] (https://github.com/adessoAG/budgeteer/files/5486795/budgeteer.-.missing.GROUP.BY.log)
attached. Patch submitted below.

Additionally, in case a person has no default daily rate, the table must show an empty cell but shows a default rate of zero (0,00).

FrederikSchlemmer commented:
@tinne I added the patch for the SQL errors.
Furthermore, I changed from a MoneyLabel to a Label. The MoneyLabel changes null values to 0,00.
That's why a normal label is needed in this case.
